### PR TITLE
WinPB: Add Installation of VS2022 Redists

### LINKED ISSUE #3772: Extract vs2022_redist_14.40.33807_10.0.26100.0.zip onto Adoptium Windows nodes via ansible
Please can we get the following Windows vs2022 redist zip extracted onto the Adoptium Windows "build" nodes (x64, x86-32 and aarch64)
unzipped into location c:\openjdk\devkit\vs2022_redist_14.40.33807_10.0.26100.0

https://github.com/adoptium/devkit-binaries/releases/download/vs2022_redist_14.40.33807_10.0.26100.0/vs2022_redist_14.40.33807_10.0.26100.0.zip

The sha256 for this zip is a29ada15d941a7b2065e9a4273fd6b97df44d089ed2b9f860ded442f7fe69767

### PR DESCRIPTION
**Pull request by steelhead31**
Fixes #3772 

Required as part of https://github.com/adoptium/temurin-build/pull/3981/files 

##### Checklist

- [X] commit message has one of the [standard prefixes](https://github.com/adoptium/infrastructure/blob/master/CONTRIBUTING.md#commit-messages)
- [X] playbook changes run through [VPC](https://ci.adoptium.net/view/Tooling/job/VagrantPlaybookCheck/) or [QPC](https://ci.adoptium.net/view/Tooling/job/QEMUPlaybookCheck/) (if you have access)

VPC Successful : https://ci.adoptium.net/job/VagrantPlaybookCheck/OS=Win2022,label=vagrant/1982/console

### CODE REVIEW

**Review by karianna — APPROVED** [review]

**Review by karianna — CHANGES_REQUESTED** [review]
Can we extract the version # to a const at the top and reference that?

**Review by karianna — APPROVED** [review]

File: ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/MSVS_2022_REDIST/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    wsdk_checksum: "a29ada15d941a7b2065e9a4273fd6b97df44d089ed2b9f860ded442f7fe69767"
+  tags: MSVS_2022_REDIST
+
+- name: Test if VS 2022 Redists Are installed
+  win_stat:
+    path: 'c:\openjdk\devkit\vs2022_redist_{{ wsdk_version }}'
+  register: vs2022sdk_installed
+  tags: MSVS_2022_REDIST
+
+- name: Check if the directory exists
```
Comment (sxa) [inline ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/MSVS_2022_REDIST/tasks/main.yml:21]:
```suggestion
- name: Check if C:\openjdk\devkit exists
```

File: ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/MSVS_2022_REDIST/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  ansible.windows.win_stat:
+    path: 'c:\openjdk\devkit'
+  register: directory_status
+  tags: MSVS_2022_REDIST
+
+- name: Create the directory if it does not exist
```
Comment (sxa) [inline ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/MSVS_2022_REDIST/tasks/main.yml:27]:
```suggestion
- name: Create  C:\openjdk\devkit if it does not exist
```

File: ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/MSVS_2022_REDIST/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  tags: MSVS_2022_REDIST
+
+- name: Unzip Visual Studio 2022 Redists
+  win_unzip:
+    src: 'c:\openjdk\devkit\vs2022_redist_{{ wsdk_version }}.zip'
+    dest: 'c:\openjdk\devkit\vs2022_redist_{{ wsdk_version }}'
+  when: not vs2022sdk_installed.stat.exists
+  tags: MSVS_2022_REDIST
+
+- name: Remove VS2022 SDK Download
```
Comment (sxa) [inline ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/MSVS_2022_REDIST/tasks/main.yml:52]:
```suggestion
- name: Remove VS2022 redists Download
```

**Review by sxa — CHANGES_REQUESTED** [review]

**Review by andrew-m-leonard — APPROVED** [review]
looks good thanks

**Review by sxa — APPROVED** [review]

**Pull request merged**